### PR TITLE
dancing-script: init at 2.0

### PR DESCRIPTION
**Pull request by wdavidw**
The [Dancing Script](https://github.com/impallari/DancingScript) font is a lively casual script where the letters bounce and change size slightly. Caps are big, and goes below the baseline.

Notes, version is coming from the .glyphs source file and is called 
2.000 in the readme 
https://raw.githubusercontent.com/impallari/DancingScript/master/sources/DancingScript.glyphs

###### Description of changes

This is a new module to integrate the Dancing Script font.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -13215,6 +13215,13 @@
     githubId = 34962284;
     name = "wchresta";
   };
+
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:13218]:
```suggestion
```

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -2843,6 +2843,8 @@ with pkgs;
   dale = callPackage ../development/compilers/dale { };
 
   damon = callPackage ../tools/admin/damon { };
+
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:2846]:
```suggestion

```

File: pkgs/data/fonts/dancing-script/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+  src = fetchFromGitHub {
+    owner = "impallari";
+    repo = "DancingScript";
+    rev = "f7f54bc1b8836601dae8696666bfacd306f77e34";
+    sha256 = "B9oAZFPH3dG/Nt5FfKfFVJYtfUKGK0AXNkQHRC7IgdU=";
+  };
+
+  installPhase = ''
+    install -m444 -Dt $out/share/fonts/truetype fonts/ttf/*.ttf
```
Comment (SuperSandro2000) [inline pkgs/data/fonts/dancing-script/default.nix:15]: If this is just the font we can also only use fetchFromGitHub and postFetch.

Comment (wdavidw) [reply pkgs/data/fonts/dancing-script/default.nix:16]: I was kind of questioning myself whether I should explore this approach. Thanks to your comment, I understand it was not such a bad idea and found an example in the source code as an example.

**Comment by wdavidw** [comment]
Thank you @SuperSandro2000 for the review. I think we shall be good now.

**Pull request merged**